tests/common/mod.rs

```diff
@@ -0,0 +1,283 @@
+    let lnd_download_endpoint = format!("{lnd_releases_endpoint}/{}", lnd_download_filename());
+
+    let resp = minreq::get(&lnd_download_endpoint).send().unwrap();
+    assert_eq!(
+        resp.status_code, 200,
+        "url {lnd_download_endpoint} didn't return 200"
+    );
+    let content = resp.as_bytes();
+
+    let decoder = GzDecoder::new(&content[..]);
+    let mut archive = Archive::new(decoder);
+    archive.unpack(lnd_dir).expect("unable to unpack lnd files");
+
+    temp_dir
+}
+
+// LndNode holds the tools we need to interact with a Lightning node.
+#[derive(Debug)]
+pub struct LndNode {
+    address: String,
+    _lnd_exe_dir: TempDir,
+    _lnd_dir_tmp: TempDir,
+    cert_path: String,
+    _handle: Child,
+}
+
+impl LndNode {
+    fn new(bitcoind_connect_params: ConnectParams, lnd_exe_dir: TempDir) -> LndNode {
+        env::set_current_dir(lnd_exe_dir.path().join("lnd-linux-amd64-v0.16.2-beta"))
+            .expect("couldn't set current directory");
+
+        let lnd_dir_binding = Builder::new().prefix("lnd").tempdir().unwrap();
+        let lnd_dir = lnd_dir_binding.path();
+
+        let cert_path = lnd_dir.join("tls.cert").to_str().unwrap().to_string();
+
+        // Have node run on a randomly assigned grpc port. That way, if we run more than one lnd node, they won't
+        // clash.
+        let port = bitcoind::get_available_port().unwrap();
+        let rpc_addr = format!("localhost:{}", port);
+        let args = [
+            format!("--rpclisten={}", rpc_addr),
+            format!("--norest"),
+            format!("--bitcoin.active"),
+            format!("--bitcoin.node=bitcoind"),
+            format!("--bitcoin.regtest"),
+            format!("--lnddir={}", lnd_dir.display()),
+            format!(
+                "--bitcoind.rpccookie={}",
+                bitcoind_connect_params.cookie_file.display()
+            ),
+            format!("--bitcoind.zmqpubrawblock=tcp://127.0.0.1:28332"),
+            format!("--bitcoind.zmqpubrawtx=tcp://127.0.0.1:28333"),
+            format!(
+                "--bitcoind.rpchost={:?}",
+                bitcoind_connect_params.rpc_socket
+            ),
+        ];
+
+        let cmd = Command::new("./lnd")
+            .args(args)
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("Failed to execute lnd command");
+
+        LndNode {
+            address: format!("https://{}", rpc_addr),
+            _lnd_exe_dir: lnd_exe_dir,
+            _lnd_dir_tmp: lnd_dir_binding,
+            cert_path: cert_path,
+            _handle: cmd,
+        }
+    }
+
+    // Setup the client we need to interact with the LND node.
+    async fn setup_client(&self) {
+        // We need to give lnd some time to start up before we'll be able to interact with it via the client.
+        let mut retry = false;
+        let mut retry_num = 0;
+        while retry_num == 0 || retry {
```

Comment:
Good call :+1:

Reply:
Actually, taking another look at this, I think this won't quite work because SubscribeState requires a tls cert parameter, which we're using the retry while loop to wait for the creation of here.